src/compile/model.ts

```diff
@@ -170,7 +171,7 @@ export abstract class Model {
 
     // If name is not provided, always use parent's givenName to avoid name conflicts.
     this.name = spec.name || parentGivenName;
-    this.title = spec.title;
+    this.title = isString(spec.title) ? {text: spec.title} : spec.title;
```

Comment:
My guess is that this test isn't for numeric titles but rather a `spec.title` that can be either a [`string | TitleParams`](https://github.com/vega/vega-lite/pull/2876/files#diff-f0bc433010488ea7f9df55b35517b91bR39)